### ISSUE TITLE
freeze and kill TB when I use "search and export"

### ISSUE THREAD

**Issue by aribbada** [thundernest/import-export-tools-ng#414]
Big problem with latest addon update, freeze and kill TB when I use "search and export" and select export ALL as single text file

**Comment by metaist** [comment]
FYI - Other folks seem to be having this situation (see #412).

**Comment by aribbada** [comment]
Yes, you're right. Me I used only to export all mails as a single text file from "search and export" option.

**Comment by cleidigh** [comment]
@aribbada 
@metaist 
Sorry for the issues all.
Quick questions before I dig into this.
OS and Thunderbird versions.
Also does downgrading to 12.0.2 restore functionality? There have been several TB updates and just want to see if that's potentially an issue.
@cleidigh

**Comment by aribbada** [comment]
I use TB version 102.11.2 (32-bit) on: win 10, win 8.1 and win server 2022 (64-bit). Same TB version and got same problem with all OS.

**Comment by metaist** [comment]
```text
$ uname -a
Linux muon 5.15.0-73-generic #80-Ubuntu SMP Mon May 15 15:18:26 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

$ thunderbird --version
 Thunderbird 102.11.0
```
Downgrading to 12.0.2 restores functionality.

**Comment by aribbada** [comment]
And yes, downgrading to 12.0.2 restores functionality.

**Comment by cleidigh** [comment]
@metaist 
@aribbada 
Thanks, so not os or TB dependent.
My virtual folder fix must have interference.
I am working on related change requirements for TB 115, I will take a break to look at these.
@cleidigh

**Comment by cleidigh** [comment]
@metaist 
@aribbada 
Yup, found it.
Should have beta by tomorrow.
@cleidigh

**Comment by cleidigh** [comment]
@metaist 
@aribbada 
This b1 should do the trick.
Doing too much with the 115 update.
https://github.com/thundernest/import-export-tools-ng/blob/v12.0.4/xpi/beta/import-export-tools-ng-12.0.4-b1-tb.xpi

@cleidigh

**Comment by aribbada** [comment]
Same problem. Emails are not exported. I pull "export" and nothing happen. The only difference is TB did not crush.

**Comment by cleidigh** [comment]
@aribbada 
Are you saying my b1 beta still didn't solve the problem?
@cleidigh

**Comment by aribbada** [comment]
> 

No. Problem still persist.

**Comment by cleidigh** [comment]
@aribbada 
Ok that's odd given the targeted fix and ability to recreate your error and subsequent fix.
I will have to add debug to see if we can get view into the issue.
@cleidigh

**Comment by cleidigh** [comment]
@metaist 
Were you able to test b1?
@cleidigh

**Comment by metaist** [comment]
Just tried it and can confirm that b1 works for me.

**Comment by cleidigh** [comment]
@metaist 
Excellent, well 1 of 2, now have to figure out why @aribbada still has an issue...
@cleidigh

**Comment by cleidigh** [comment]
@aribbada 
I forgot to ask if you can check the console output for any errors?
@cleidigh

**Comment by aribbada** [comment]
> I forgot to ask if you can check the console output for

I'm not so technical..teach me how to do the console output

**Comment by aribbada** [comment]
[https://i.gyazo.com/faa5ef4ff1ae99227f196da41ca2294c.png](url) this is what i'm doing. And click on Export and nothing happens. with 12.0.2 work fine, all emails are exported. But with 12.0.4-b1 nothing happen.

**Comment by aribbada** [comment]
Sorry, I click on "Export" and choose the folder to export and after that nothing happen with 12.0.4-b1.

**Comment by cleidigh** [comment]
@aribbada 
To open and check the console use Control -Shift-J
Do this from the main mail view before doing anything else. Then in the console click the trash can to clear. Go back to the mail window, now try the search and export. Go back and check if the log has any errors.
If we don't see anything I will have to add trace to the next beta.
@cleidigh

**Comment by aribbada** [comment]
https://i.gyazo.com/bd57889f8d84253120c673199ffa3999.png

**Comment by cleidigh** [comment]
@aribbada 
Ok thanks that error is past the area where I have my fix so we are looking at perhaps something different. Was this exporting to a single text file?
Can you try to export to EML and also individual text files? If they don't work check for errors.
@cleidigh

**Comment by aribbada** [comment]
This is for all in EML format: https://i.gyazo.com/e27a3d043ecaeb79ae737c4312a22e93.png

For plain text: https://i.gyazo.com/29330434bcc238187c6fa755f066cf4c.png

Still not work, not export.

**Comment by cleidigh** [comment]
@aribbada 
Thanks.
This shows they are all failing on the same method call, however, I don't have enough information why.
One last test and then I will have to do another beta.
Can you take the first message that comes up in the search and try to do a single message export of that. Find it, right click and save the message as eml. That eliminates the search, but uses the same export mechanism.
@cleidigh

**Comment by aribbada** [comment]
Yes, no error on that. Export for first eml was successfully and no errors on console. Tried with two and works. Exported eml or text without problems. Also what I tried and works: checked export "selected" option and with CTRL+A I selected all 7280 emails and after that I've clicked on "export" and works !! I think the problem is when the "All" option is selected(checked).

https://i.gyazo.com/d4b86e7ed7275c6907f59aea5ae263d2.png

**Comment by cleidigh** [comment]
@aribbada 
Ok so to be clear, if your search has 7280 results and you choose export selected, highlight all, then export, all are exported. If you take the same 7280 and just try export all, this fails. If so then I can focus debug on that.
@cleidigh

**Comment by aribbada** [comment]
Yes, correct.

**Comment by cleidigh** [comment]
ok, now I can look at that. Will try to do beta for tomorrow.
Thanks for the testing.
@cleidigh

**Comment by aribbada** [comment]
Thank you for your efforts.

**Comment by cleidigh** [comment]
@aribbada 
@metaist 
I have added some debug, but more importantly I think I found an issue that affects the  total exported messages.
Give it a shot and will debug as needed.
https://github.com/thundernest/import-export-tools-ng/blob/v12.0.4/xpi/beta/import-export-tools-ng-12.0.4-b2-tb.xpi
@cleidigh

**Comment by aribbada** [comment]
Test done and this version work fine!!! Many thanks!

**Comment by Yankeese41** [comment]
I have the same issue. I usually export by date in 1-month increments. I have always used "Export All" with no problem. Yesterday Tbird froze as described above. What I did was choose "Export Selected", selected all the search results, then exported. That worked. It is a usable workaround until the issue is fixed.

**Comment by cleidigh** [comment]
@Yankeese41 
Great, you got the beta, was going to point.
@cleidigh

**Comment by cleidigh** [comment]
@metaist 
@aribbada 
@Yankeese41 
Version 12.0.4 with this patch was just released so check and update if not done already.
Cheers 
Christopher

**Comment by metaist** [comment]
Confirming this works for me. Thank you!
